CodeEdit/Utils/Extensions/NSApplication/NSApp+openWindow.swift

```diff
@@ -6,30 +6,15 @@
 //
 
 import AppKit
+import SwiftUI
 
-extension NSApplication {
-    func openWindow(_ id: SceneID) {
-        NSMenuItem.openWindowAction = { (id, nil) }
-        openWindowPerform()
-    }
-
-    func openWindow(_ id: SceneID, value: any Codable & Hashable) {
-        NSMenuItem.openWindowAction = { (id, value) }
-        openWindowPerform()
-    }
-
-    func openWindow(value: any Codable & Hashable) {
-        NSMenuItem.openWindowAction = { (nil, value) }
-        openWindowPerform()
-    }
-
-    private func openWindowPerform() {
-        let item = NSApp.windowsMenu?.items.first { $0.title == "OpenWindowAction" }
-        if let item, let action = item.action {
-            NSApp.sendAction(action, to: item.representedObject, from: nil)
-        }
+extension OpenWindowAction {
+    func callAsFunction(sceneID: SceneID) {
```

Comment:
Can or should we do something like this in order to avoid changing all of the other uses of `openWindow` to include `sceneID`?

```suggestion
    func callAsFunction(_ sceneID: SceneID) {
```

Reply:
If we omit the parameter name, we still have to change other uses because the previous code uses `openWindow(id:)`. We can overload it like `func callAsFunction(id sceneID: SceneID)` but I don't prefer that because `sceneID` is more descriptive.